### PR TITLE
Comparison operators in template instantiations

### PR DESCRIPTION
**Pull request by ansharlubis**
Based on the example sent in #2027, I made necessary modifications to have the example working.

Note:
- Need to clean up `visit_Instantiate` when handling intrinsic operators.

### CODE REVIEW

File: src/lfortran/semantics/ast_symboltable_visitor.cpp [hunk 1]
```diff
@@ -2447,30 +2476,63 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
...
-                            value = ASRUtils::EXPR(ASR::make_IntegerBinOp_t(al, x.base.base.loc,
-                                lexpr, op, rexpr, ASRUtils::duplicate_type(al, ltype), nullptr));
+                            if (is_binop) {
+                              value = ASRUtils::EXPR(ASR::make_IntegerBinOp_t(al, x.base.base.loc,
+                                  lexpr, binop, rexpr, ASRUtils::duplicate_type(al, ltype), nullptr));
+                            } else if (is_cmop) {
+                              value = ASRUtils::EXPR(ASR::make_IntegerCompare_t(al, x.base.base.loc,
+                                  lexpr, cmop, rexpr, 
+                                  ASRUtils::TYPE(ASR::make_Logical_t(al, x.base.base.loc, 4)), nullptr));
+                            }
```
Comment (certik) [inline src/lfortran/semantics/ast_symboltable_visitor.cpp:2498]: What happens if it is neither? Should we give some exception or at least an assert?

**Review by certik — APPROVED** [review]
This is fine to merge as is, I left a minor comment.

@ansharlubis go ahead and merge it, unless you want to still improve something.

**Comment by certik** [comment]
I am just going to merge it, so that it's in. We can improve upon it later.

File: src/lfortran/semantics/ast_symboltable_visitor.cpp (hunk 1 above)
Comment (ansharlubis) [reply src/lfortran/semantics/ast_symboltable_visitor.cpp:2498]: The implementation is not that good right now, I want to add asserts and separate the treatment for binary and comparison operators later.

**Pull request merged**